paddle/fluid/operators/elementwise_op_function.h

```diff
@@ -468,15 +483,22 @@ static __global__ void ElemwiseGradBroadcast2CUDAKernel(
   }
 
   if (dy) {
+    T val = shm[threadIdx.x];
+    int warpSize = 32;
```

Comment:
`warpSize` is a global variable in CUDA. It just likes blockDim. `warpSize` may not be 32 in some devices.

Reply:
Maybe, but it seems that warpSize cannot be got from kernel function. We can get it by calling `deviceProp.warpSize`, and pass it to `reduceSum`.